dialogue-core/src/main/java/com/palantir/dialogue/core/Config.java

```diff
@@ -81,4 +137,8 @@ default void check() {
                     SafeArg.of("numUris", rawConfig().uris().size()));
         }
     }
+
+    static String stripMeshPrefix(String input) {
+        return input.startsWith(MESH_PREFIX) ? input.substring(MESH_PREFIX.length()) : input;
```

Comment:
the apache client allows custom protocol handlers, so we can avoid any url rewriting, and unsafe messages will include the mesh prefix for easier debugging. wdyt?

Reply:
So i guess this decision kinda impacts whether this `mesh-` flag should be allowed to implement stuff inside whatever 'ChannelFactory' people supply, or whether it's a purely dialogue-internal thing.

Personally, I'd prefer to keep this dialogue internal, because if we're asking more people to make decisions based on this then I think we should find a way of exposing it in a nice typed way.